BizHawk.Client.Common/movie/tasproj/TasStateManager.cs

```diff
@@ -453,110 +273,114 @@ public void Load(BinaryReader br)
 		public KeyValuePair<int, byte[]> GetStateClosestToFrame(int frame)
 		{
 			var s = _states.LastOrDefault(state => state.Key < frame);
+			if (s.Key > 0)
+			{
+				return s;
+			}
 
-			return this[s.Key];
+			return new KeyValuePair<int, byte[]>(0, InitialState);
 		}
 
-		/// <summary>
-		/// Returns index of the state right above the given frame
-		/// </summary>
 		public int GetStateIndexByFrame(int frame)
 		{
 			return _states.IndexOfKey(GetStateClosestToFrame(frame).Key);
 		}
 
-		/// <summary>
-		/// Returns frame of the state at the given index
-		/// </summary>
 		public int GetStateFrameByIndex(int index)
 		{
-			// feos: this is called super often by decay
-			// this method is hundred times faster than _states.ElementAt(index).Key
 			return _states.Keys[index];
 		}
 
-		private ulong _used;
-		private ulong Used
+		private bool IsMarkerState(int frame)

```

Comment:
So it survived in the end?

Reply:
As a private method within the specific IStateManager implementation, yes.  Things like this are perfectively fine.  It did not make it into the interface however.